utils/src/main/java/org/nervos/ckb/crypto/Blake2b.java

```diff
@@ -0,0 +1,43 @@
+package org.nervos.ckb.crypto;
+
+import org.bouncycastle.crypto.digests.Blake2bDigest;
+import org.nervos.ckb.utils.Numeric;
+
+/**
+ * Created by duanyytop on 2019-03-08.
+ * Copyright © 2018 Nervos Foundation. All rights reserved.
+ */
+public class Blake2b {
+
+    private Blake2bDigest blake2bDigest;
+
+    public static Blake2b getInstance() {
```

Comment:
Why do we need to hold a static instance? Wouldn't using `new Blake2b()` directly make more sense?

Reply:
This keeps the static nature of the previous implementation thus is still not type safe. For example, if we change the test's last part to:

```java
blake2b.update("The quick brown fox jumps over the lazy dog".getBytes(StandardCharsets.UTF_8));
Blake2b anotherBlake2b = Blake2b.getInstance();
anotherBlake2b.update("The quick brown fox jumps over the lazy dog".getBytes(StandardCharsets.UTF_8));
assertThat(blake2b.doFinalString(), is("0xabfa2c08d62f6f567d088d6ba41d3bbbb9a45c241a8e3789ef39700060b5cee2"))
```

I'm pretty positive the test would fail.

Reply:
I think twice that static instance do cause security risk. And the test case above would fail. I refactor with normal class instance.

Reply:
This problem is my owing consideration.

Reply:
> I think twice that static instance do cause security risk. And the test case above would fail. I refactor with normal class instance.

BTW now that we refactored out the static instance, we should also remove the holder class (if not already).

Reply:
holder class? I don't quite understand.

Reply:
I meant the `Blake2bHolder` class - which was apparently already removed 👍.

Reply:
Yes. It is just a single instance implement method.